wasm-runtime/std/src/lib.rs

```diff
@@ -139,12 +153,18 @@ macro_rules! impl_stubs {
 			$(
 				#[no_mangle]
 				pub fn $name(input_data: *mut u8, input_len: usize) -> u64 {
-					let input = unsafe {
-						$crate::vec::Vec::from_raw_parts(input_data, input_len, input_len)
+					let input = if input_len == 0 {
+						$crate::vec::Vec::new()
+					} else {
+						unsafe {
+							$crate::vec::Vec::from_raw_parts(input_data, input_len, input_len)
```

Comment:
as I said earlier in the chat, this will take ownership of this allocation and `Vec` will drop it at the end of the method. And if for some reason this pointer will be deallocated in other way it was deallocated then some bad things might happen.

I think that the best way to do this is to make client code responsible for deallocating `input_data` pointer (because the client code allocated this pointer in a first place) and use slice instead of a vector.

If this isn't desired for some reason, then I think it would be better to use slice and deallocate this pointer explicitly, preferably with a comment